sdk/identity/azure-identity/src/main/java/com/azure/identity/implementation/IdentityClient.java

```diff
@@ -1336,8 +1341,7 @@ private HttpPipeline setupPipeline(HttpClient httpClient) {
                 URL url = null;
                 HttpURLConnection connection = null;
                 try {
-                    url =
-                            new URL(String.format("%s?%s", endpoint, payload.toString()));
+                    url = new URL(String.format("%s?%s", endpoint, payload));
```

Comment:
Long term we should look into replacing simple String.formats with string concatenation, in most cases it will be easier to read but also be better performance wise as String.format needs to parse the formattable string to determine replacement locations and their handling